.github/workflows/docs.yml

```diff
@@ -5,12 +5,26 @@ on:
     branches:
       - 'main'
 
+  # Allows running the action manually.
+  workflow_dispatch:
+
 env:
   CARGO_TERM_COLOR: always
   RUSTDOCFLAGS: --html-in-header header.html
 
+# Sets the permissions to allow deploying to Github pages.
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Only allow one deployment to run at a time, however it will not cancel in-progress runs.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
```

Comment:
why this choice? Cancelling would make sense here, we only care about the last version

Reply:
In the example for using pages artifacts, the following comment is provided about the concurrency setting:

```yaml
# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
```

My best guess is that perhaps cancelling the deploy job while it's still uploading to Github pages could be bad, but I've never tested it. ([See docs on concurrency.](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency))

I think it would be best to leave this. The developer docs don't seem like a time-sensitive thing. If this is a risk of messing up Github pages, I don't think it's worth the couple of extra minutes saved.

Reply:
Not a fan of adding that for now, but it doesn't really matter